### DISCUSSION TITLE
API for determining Personal Access Token scopes?

### DISCUSSION THREAD

**Discussion by hwine** [community/community#24345] [Apps, API and Webhooks]
<p>Is there a way to retrieve the current scopes for a Personal Access Token (PAT)?</p>
<p><strong>Use Case</strong> :</p>
<p>As an organization owner, I want to know the scopes currently granted to a PAT for a member of my organization that has been leaked.</p>
<p>Thanks,</p>
<p>–Hal</p>

**Comment by hwine** [comment, marked as answer]
<p>Got the answer from Ivan support:</p>
<blockquote>
<p>If it hadn’t been revoked, you could have figured our which scopes it has by making any API call with that token (<a href="https://developer.github.com/v3/#authentication" rel="nofollow noopener">https://developer.github.com/v3/#authentication</a>) and then looking at the X-OAuth-Scopes response header. That header tells you which scopes the token has.</p>
</blockquote>
<p>I’ve spent so much time staring at JSON that I forgot about the HTTP headers. /o\</p>
<p>Thanks, Ivan!</p>

**Comment by yakov116** [comment]
<p>The issue is that if you change the scope of the API after creating it the headers do not update!</p>

**Comment by hwine** [comment]
<p><a class="mention" href="/yakov116">@yakov116</a> – if you’re seeing that behavior, that would be a separate bug that should be reported. I rarely change scopes, so have not encountered this.</p>
